sdk/core/azure-core/src/main/java/com/azure/core/util/polling/ChainedPollingStrategy.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple2;
+import reactor.util.function.Tuples;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A polling strategy that chains multiple polling strategies, finds the first strategy that can poll the current
+ * long running operation, and polls with that strategy.
+ *
+ * @param <T> the type of the response type from a polling call, or BinaryData if raw response body should be kept
+ * @param <U> the type of the final result object to deserialize into, or BinaryData if raw response body should be
+ *           kept
+ */
+public class ChainedPollingStrategy<T, U> implements PollingStrategy<T, U> {
+    private final List<PollingStrategy<T, U>> pollingStrategies;
+    private volatile PollingStrategy<T, U> pollableStrategy = null;
```

Comment:
Based on the [comment in PollingStrategy interface](https://github.com/Azure/azure-sdk-for-java/pull/22795/files#diff-728313447171991d15f593665b1a20acddf9a1c8632d9967b3ac9eb5f31f7b3aR19), do we need this to be volatile?

Reply:
removed